Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -143,8 +121,15 @@ import Foundation
         if let clientToken {
             configPath = clientToken.configURL.absoluteString
         }
+        
+        let authCredential = clientToken?.authorizationFingerprint ?? tokenizationKey
+        if let authCredential,
```

Comment:
I acknowledge this is a little ugly, but would rather tackle it in another PR - to move away from _both_ the clientToken & tokenizationKey being optional and leverage the `BTAPIClientAuthorization` enum with an associated string value instead.

This PR has a lot of change, though so I am opting to do it later

Reply:
Do we think for now we could clean this up slightly by using a `guard` here:
```swift
guard let authorization = clientToken?.authorizationFingerprint ?? tokenizationKey else {
    completion(nil, BTAPIClientError.invalidAuthorization)
    return
}
```